tests/test_fits.py

```diff
@@ -14,6 +14,25 @@
 my_max = 1000  # Typically number of galaxies in loop
 except_int = 750  # Where test exception occurs in loop
 filename = "MyFile.Fits"
+cls_file = "Cls.npz"
+
+
+def test_read_write_cls(rng: np.random.Generator, tmp_path: pathlib.Path) -> None:
+    cls = rng.normal(size=(10, 10))
+    save_cls(tmp_path / cls_file, cls)
+
+    assert pathlib.Path.exists(tmp_path / cls_file)
+
+    with np.load(tmp_path / cls_file) as npz:
+        values = npz["values"]
+        split = npz["split"]
+
+    np.testing.assert_allclose(values, np.concatenate(cls))
+    np.testing.assert_allclose(split, np.cumsum([len(cl) for cl in cls[:-1]]))
+    np.testing.assert_allclose(cls, np.split(values, split))
+
+    npz = load_cls(tmp_path / cls_file)
+    np.testing.assert_allclose(npz, cls)
```

Comment:
I could not find a better way of testing these functions in 2 independent tests. I could do that if I don't use the `tmp_path` fixture. Both cases have pros and cons, but I don't have any strong opinions.

Reply:
I think this is fine, we aren't testing that we adhere to an external file format here, so we don't need to test that `numpy.savez()` doesn't change.